assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -125,6 +129,372 @@ export default {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixType() {
+              if (this.channel.chatable.users.length === 1) {
+                return "image";
+              } else {
+                return "text";
+              }
+            }
+
+            get prefixValue() {
+              if (this.channel.chatable.users.length === 1) {
+                return avatarUrl(
+                  this.channel.chatable.users[0].avatar_template,
+                  "tiny"
+                );
+              } else {
+                return this.channel.chatable.users.length;
+              }
+            }
+
+            get prefixCSSClass() {
+              const activeUsers = this.chatService.presenceChannel.users;
+              const user = this.channel.chatable.users[0];
+              if (
+                !!activeUsers?.findBy("id", user?.id) ||
+                !!activeUsers?.findBy("username", user?.username)
+              ) {
+                return "active";
+              }
+              return "";
+            }
+
+            get suffixType() {
+              return "icon";
+            }
+
+            get suffixValue() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCSSClass() {
+              return "urgent";
+            }
+
+            get hoverType() {
+              return "icon";
+            }
+
+            get hoverValue() {
+              return "times";
+            }
+
+            get hoverAction() {
+              return () => {
+                this.chatService.unfollowChannel(this.channel);
+              };
+            }
+
+            get hoverTitle() {
+              return I18n.t("chat.direct_messages.leave");
+            }
+
+            get currentWhen() {
+              return (
+                this.chatService.router.currentRouteName?.startsWith(
+                  "chat.channel"
+                ) && this.channel.id === this.chatService.activeChannel?.id
+              );
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = [];
+
+            constructor() {
+              super(...arguments);
+
+              this.chatService = container.lookup("service:chat");
+              this.sidebar.appEvents.on(
+                "chat:refresh-channels",
+                this._refreshPms
+              );
+              this.sidebar.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshPms
+              );
+              this._refreshPms();
+            }
+
+            teardown() {
```

Comment:
```suggestion
            willDestory() {
```

I removed the `teardown` function in https://github.com/discourse/discourse/pull/17550